package.json

```diff
@@ -0,0 +1,74 @@
+{
+  "name": "fastify-passport",
+  "version": "0.0.1",
+  "description": "Simple, unobtrusive authentication for Fastify.",
+  "main": "dist/src/index.js",
```

Comment:
Why the compiled files ends in `dist/src`, isn't this supposed to be `dist/`?

Reply:
It could be, but I did it a little differently. This project builds both the `src` and the `test` folders when building so that building fails if the tests don't compile too. That way, it's safe to run the tests with typescript doing `transpileOnly`, which is fast for dev iteration cycles, but we still check the typecheck the tests at some point, so CI will fail if you write broken typescript in the tests.

Reply:
OK. Maybe we can try a different approach, in order to be sure that dist folder contains only the code that is going to be published to npm. We can:
- `test` folder isn't compiled anywhere, but it is checked for types
- before running the tests, `src` folder is compiled
- tests import the package from `dist`folder (using the main field of package.json): this ensures that even the generated JS is what we expect and what the user is going to use (sometimes typescript code is passing the test but the generated `.js` + `d.ts` files are not correct).
- dev test script watches `src` and recompilers only the file that is needed to be compiled and the test script watches the `dist` folder.

Let me know what do you think. I really don't like the idea of having `src` folder inside `dist` since it contains `.js` files that are not src at all.

Reply:
I know what you mean, it does seem a little wonky, but I still prefer the approach as it is now. I think having to build the whole library before running a test is super annoying and ended up here because I was tired of staring at an empty terminal. I prioritized developer speed over filesystem layout. `jest`'s TypeScript support uses babel with the `transpileOnly` flag set to get JS for running as fast as possible, without type-checking, to keep the iteration cycle short and I really like that. I believe it also already has the only-recompile-what-has-changed behaviour because of Babel's built in caching. 

I also think having to run a watcher process should be avoided if possible -- new developers who want to contribute should just be able to run `yarn test` and have it just work instead of having to dive into the `scripts` to see how it is "supposed" to be done. 

I renamed the folder to `build` because I think that's a bit clearer and less wonky when containing `src` and `test`, and also the `package.json` only includes `build/src` for publishing so the tests aren't published to npm.

Reply:
I think that would work, but, is it worth the added complexity just to change the folder structure? What's the downside of having compiled tests in the `build` folder? If you feel super strongly about it I don't mind changing things.

Reply:
I want to use this file structure for `fastify-plugin-ts` template to add to the `fastify-cli`. I want to make it "simple", easy to understand, and structured like the common TS projects are. 

If you don't mind it will be amazing to make this change. If structured like that, we can also add more starters for other test runners (if someone prefers tap/mocha over jest, for example). :)

Reply:
Ok, I think I found a good middle path. I forgot I was already using `ts-jest` instead of `babel-jest` to get project-wide typechecking, which is necessary for using declaration merging on `FastifyRequest` and whatnot. So, typechecking was actually already running when running the tests 😓 

`tsc` will compile stuff outside the `include` config if there is no `outDir` set, which means `ts-jest` won't complain about compiling the tests. It would if `outDir` was set, which is what I had before and why I ended up butting both in `dist`. With `outDir` unset though, we can build just the src into `dist` and the containing folder gets stripped because no tests are included. 

This is the same setup that `tsdx` uses so I think it's fair to say it is structured in a commonly understood way. Sound ok?

Reply:
If I get this correctly, it absoultely makes sense! Go for it!